pylmod/base.py

```diff
@@ -48,82 +45,84 @@ def __init__(
         self.cert = cert
 
         self.urlbase = urlbase
+        if not urlbase.endswith('/'):
+            self.urlbase += '/'
         self.ses = requests.Session()
         self.ses.cert = cert
         self.ses.timeout = self.TIMEOUT  # connection timeout
         self.ses.verify = True  # verify site certificate
+        # Mount the retry adapter to the base url
+        self.ses.mount(urlbase, HTTPAdapter(max_retries=self.RETRIES))
 
         log.debug("------------------------------------------------------")
-        log.info("[PyLmod] init base=%s", urlbase)
+        log.info("[PyLmod] init urlbase=%s", urlbase)
+
+    @staticmethod
+    def _data_to_json(data):
+        """Convert to json if it isn't already a string"""
+        if type(data) not in [str, unicode]:
+            data = json.dumps(data)
+        return data
+
+    def _url_format(self, service):
+        """Generate URL from urlbase, service
+
+        Args:
+            service (str): The endpoint service to use, i.e. gradebook
```

Comment:
that actually is the sub service to gradebook in LMod, it is `assignment`, `assignments`, `gradebook`, or `academicterm`